src/main/java/org/folio/circulation/infrastructure/storage/users/UserRepository.java

```diff
@@ -93,6 +93,13 @@ public UserRepository(Clients clients) {
         loan -> loan.withUser(users.getOrDefault(loan.getUserId(), null)))));
   }
 
+  public CompletableFuture<Result<Collection<Loan>>> findUsersForLoans(Collection<Loan> loans) {
+    return getUsersForLoans(loans)
+      .thenApply(r -> r.map(users -> loans.stream()
+        .map(loan -> loan.withUser(users.getOrDefault(loan.getUserId(), null)))
```

Comment:
Should we filter loans with user!=null ?

Reply:
This is not necessary: notice scheduling process is interrupted gracefully few calls down the call chain if loan has no user.